### PR TITLE
[6.3] view VDC subscription available/content products (BZ1366327)

### PR DESCRIPTION
**Pull request by ldjebran**
cover: https://bugzilla.redhat.com/show_bug.cgi?id=1366327
depend on PR (locators): https://github.com/SatelliteQE/robottelo/pull/5282
```
(sat-6.3) dl@DL:~/projects/redhat/robottelo$ pytest -v tests/foreman/ui/test_subscription.py::SubscriptionTestCase::test_positive_view_VDC_subscription_products
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dl/.pyenv/versions/sat-6.3/bin/python
cachedir: .cache
rootdir: /home/dl/projects/redhat/robottelo, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-22 15:30:46 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1378009', '1245334', '1217635', '1226425', '1147100', '1311113', '1199150', '1204686', '1221971', '1103157', '1230902', '1230865', '1214312', '1079482']

2017-09-22 15:30:46 - conftest - DEBUG - Collected 1 test cases


tests/foreman/ui/test_subscription.py::SubscriptionTestCase::test_positive_view_VDC_subscription_products <- robottelo/decorators/__init__.py PASSED

================================================== 0 tests deselected ==================================================
============================================== 1 passed in 666.90 seconds ==============================================
```

### CODE REVIEW

**Comment by ldjebran** [comment]
rebased branch after dependency https://github.com/SatelliteQE/robottelo/pull/5282 merged 
```console
(sat-6.3) dl@DL:~/projects/redhat/robottelo$ pytest -v tests/foreman/ui/test_subscription.py::SubscriptionTestCase::test_positive_view_VDC_subscription_products
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dl/.pyenv/versions/sat-6.3/bin/python
cachedir: .cache
rootdir: /home/dl/projects/redhat/robottelo, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-22 17:02:13 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1378009', '1245334', '1217635', '1226425', '1147100', '1311113', '1199150', '1204686', '1221971', '1103157', '1230902', '1230865', '1214312', '1079482']

2017-09-22 17:02:13 - conftest - DEBUG - Collected 1 test cases


tests/foreman/ui/test_subscription.py::SubscriptionTestCase::test_positive_view_VDC_subscription_products <- robottelo/decorators/__init__.py PASSED

================================================== 0 tests deselected ==================================================
============================================== 1 passed in 396.08 seconds ==============================================
```

File: robottelo/ui/subscription.py [hunk 1]
```diff
@@ -79,3 +79,19 @@ def refresh(self):
...
+    def get_provided_products(self, subscription_name):
+        """Return a list of product names provided by the subscription name"""
+        self.search_and_click(subscription_name)
+        self.click(tab_locators['subs.sub.tab_details'])
+        return [element.text
+                for element in
+                self.find_elements(locators['subs.sub.provided_products'])]
+
+    def get_content_products(self, subscription_name):
+        """Return a list of product names provided by the subscription name"""
```
Comment (oshtaier) [inline robottelo/ui/subscription.py]: seems that docstring is duplicated by copy/paste

**Comment by ldjebran** [comment]
@oshtaier updated that string, update should not affect test results

File: tests/foreman/ui/test_subscription.py [hunk 2]
```diff
@@ -228,3 +241,93 @@ def test_positive_access_with_non_admin_user_with_manifest(self):
...
+            content_view=content_view
+        ).create()
+        # add the default RH subscription
+        for sub in subscription.search():
+            if sub.read_json()['product_name'] == DEFAULT_SUBSCRIPTION_NAME:
+                activation_key.add_subscriptions(data={
+                    'quantity': 1,
+                    'subscription_id': sub.id,
+                })
+            break
```
Comment (abalakh) [inline tests/foreman/ui/test_subscription.py]: Shouldn't this `break` be placed inside `if`?

**Comment by ldjebran** [comment]
@oshtaier @abalakh comments addressed
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/ui/test_subscription.py::SubscriptionTestCase::test_positive_view_VDC_subscription_products
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-10-03 18:11:46 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_subscription.py::SubscriptionTestCase::test_positive_view_VDC_subscription_products <- robottelo/decorators/__init__.py PASSED

============================================== 1 passed in 574.57 seconds ==============================================
```

**Pull request merged**